test/tasks/suggested_resource_rake_test.rb

```diff
@@ -52,19 +52,63 @@ def setup
         Rake::Task['suggested_resources:reload'].invoke(remote_file)
       end
     end
-    assert_equal 'Some CSV columns missing: ["Phrase"]', error.message
+    assert_equal 'Some CSV columns missing: ["phrase"]', error.message
   end
 
   # assert_nothing_raised is viewed as an anti-pattern, but I'm leery of a test
   # with no assertions. As a result, we use a single assertion to confirm
   # something happened.
   test 'reload succeeds if extra columns are present' do
-    records_before = Detector::SuggestedResource.count # We have three fixtures at the moment
+    records_before = SuggestedResource.count # We have three fixtures at the moment
     VCR.use_cassette('suggested_resource:reload with extra field') do
       remote_file = 'http://static.lndo.site/suggested_resources_extra.csv'
       Rake::Task['suggested_resources:reload'].invoke(remote_file)
     end
 
-    assert_not_equal records_before, Detector::SuggestedResource.count
+    assert_not_equal records_before, SuggestedResource.count
+  end
+
+  test 'reload handles duplicate titles in CSV' do
+    records_before = SuggestedResource.count
+    VCR.use_cassette('suggested_resource:reload dup title') do
+      remote_file = 'http://static.lndo.site/suggested_resources_dup_title.csv'
+      Rake::Task['suggested_resources:reload'].invoke(remote_file)
+    end
+
+    # Confirm that one, not two, new suggested resource was created
+    assert records_before + 1, SuggestedResource.count
+  end
+
+  test 'reload handles duplicate URLs in CSV' do
+    records_before = SuggestedResource.count
+    VCR.use_cassette('suggested_resource:reload dup url') do
+      remote_file = 'http://static.lndo.site/suggested_resources_dup_urls.csv'
+      Rake::Task['suggested_resources:reload'].invoke(remote_file)
+    end
+
+    # Confirm that one, not two, new suggested resource was created
+    assert records_before + 1, SuggestedResource.count
+  end
+
+  test 'reload handles duplicate phrases in CSV' do
+    records_before = Term.count
+    VCR.use_cassette('suggested_resource:reload dup phrases') do
+      remote_file = 'http://static.lndo.site/suggested_resources_dup_phrases.csv'
+      Rake::Task['suggested_resources:reload'].invoke(remote_file)
+    end
+
+    # Confirm that one, not two, new term was created
+    assert records_before + 1, Term.count
+  end
+
+  test 'reload handles phrases that match existing terms' do
+    records_before = Term.count
+    VCR.use_cassette('suggested_resource:reload from remote csv') do
+      remote_file = 'http://static.lndo.site/suggested_resources.csv'
+      Rake::Task['suggested_resources:reload'].invoke(remote_file)
+    end
+
+    # The CSV has two rows, one of which is an existing term (web of science). So, only one term should be created.
```

Comment:
Non blocking. You asked about tests so I spent a bit more time than I normally might with these.

The only suggestion I have is that we might confirm for each of these the change in Term count _and_ the change in SuggestedResource count both match the expectations as sometimes only once changes, sometimes both.

The other comment would be that I might test the bulk_loader method more directly rather than the rake task which just adds complexity. This looks like a carry-forward from the original implementation and just updated to the new spec so I see why you chose this option for this work as changing behavior _and_ how we test is more risky so for now this is probably best to keep as-is (and maybe add assertions for SuggestedResource count if it feel it adds value to track both.

Reply:
This is a good point, and as I was looking back into this, I realized that even the assertions we _have_ might not be matching the expectations. Sigh. I want to make sure we get this right before merge, so I'm going to double-check all of these. I'll add the recommended assertions back in as part of that review.

Reply:
@JPrevost I updated the tests to include assertions on both record types, and I updated the file fixtures (and cassettes) to more accurately reflect the cases we're trying to test. I know you've already approved, but I'm going to re-request review because this feels like a significant enough change to warrant it.

Reply:
Great change, thanks!